### ISSUE TITLE
Provide a way to customize value previews in variable views

### ISSUE THREAD

**Issue by zero-plusplus** [zero-plusplus/vscode-autohotkey-debug#329]
[This new feature](https://code.visualstudio.com/updates/v1_92#_show-variable-types-while-debugging) in 1.92 allows types that could only be seen by hovering to be displayed in the value preview location (?) to be displayed.

I saw this feature and thought about how to customize the format of the value preview.

For example, it might be useful to be able to write numbers not only in decimal but also in hexadecimal, such as `123 (0x7B)`.

This is just an idea, so I don't know if it can actually be implemented. (It should probably be possible to implement it due to the specifications.)

If you have any ideas, please write them in the comments.

**Comment by SAbboushi** [comment]
Hexadecimal would be very helpful for me

**Comment by zero-plusplus** [comment]
This idea seems more useful than the #320 feature, since you can check the hexadecimal number without clicking on it.

The attribute of launch.json is going to be objects with `type` in the key and `format` in the value.

In the above example, it would be good to set `{ integer: "{value} ({hex})" }`.
The `value` here represents the original value. (It may be changed at the time of implementation, though.)

**Comment by SAbboushi** [comment]
If I understand, you are saying that we can configure which formats to display upon hover (e.g. Binary, Decimal, Hex, Scientific Notation...)?  How about ability to copy desired format value from hover?

**Comment by zero-plusplus** [comment]
I had not thought about that point.

If you use formatting to change the display, you should be able to do what you want to do because the same values will be displayed when you hover.

However, copying some of the values displayed when hovering requires the use of shortcut keys, and it is faster and more reliable to use the context menu, as they will disappear if the mouse cursor is accidentally moved outside the window.

Also, please note that this feature will only apply to variable view and will not work for others. In other words, only the actual values will be displayed when hovering over them in the editor.

**Comment by zero-plusplus** [comment]
I noticed that this feature can be integrated with #312.

At the time, I saw it as a static watch expression that is displayed conditionally, such as only when unit tests are run, but it may be better to implement the feature to convert values to a specific format here, as in this case.

The drawback is that both the grouping of variables and the rewriting of values need to be controlled by a single option, which makes the configuration too difficult for most people to handle.

**Comment by zero-plusplus** [comment]
@SAbboushi 

Sorry. I didn't have a way to contact you so I'll explain here.

I threw away the MIT license and started a new project.
This is a big backwards for my plan, so I thought I should explain it to you, who have supported this extension from the beginning.

I don't know the details, but since today is the end of the month, next month's sponsorship may be renewed tomorrow, so I contacted you to ask you to think again about whether you want to continue or not.

I apologize for the delay in explanation due to the need to consider many things.

I am prioritizing speed and minimizing review because it takes me almost an hour to post one message. Sorry if this is difficult to understand.

**Comment by SAbboushi** [comment]
Hello and my sincere apologies for taking so long to respond.

Thank you for your very considerate message.  I am sad about the situation but I understand.

Since you are planning to continue development with the hope of releasing an extension at some time in the future, I will continue my sponsorship for at least another year if that is OK with you.

I did receive your email that you sent to gi****@w3x*.com.  I would be happy if you would email me again but also provide me with your email address so we can stay in touch if you are willing.

I wish for you good success in accomplishing your objective and send my best regards-

**Comment by zero-plusplus** [comment]
No problem at all. Thanks for the reply.

> Since you are planning to continue development with the hope of releasing an extension at some time in the future, I will continue my sponsorship for at least another year if that is OK with you.

I will not force you to end your sponsorship because I respect your feelings. And I really appreciate your feelings.

The first release is just basic syntax highlighting that has already been developed in a private repository,
The work itself should be minimal.

However, the actual work time may be greater, as the mechanism needs to be enhanced to facilitate future maintenance.

You can see the my activities leading up to the release [here](https://github.com/zero-plusplus/autohotkey-devtools/commits/prototype/v0.0.1/) (It's taking me a while to commit, but I'm working on it basically every day 👍 ).

---

>  I would be happy if you would email me again but also provide me with your email address so we can stay in touch if you are willing.

I used to write my e-mail address in my profile, but I removed it a long time ago because it seemed to have more disadvantages.

So I tried to find a way to send you a private message so that you can share your email address without making it public, but github doesn't seem to offer that method, so can I send it to your account on the AutoHotkey Forums?

I will send my email address there after confirmation.

---

> I wish for you good success in accomplishing your objective and send my best regards-

Thank you so, so much. I will continue my efforts (while resting well, of course, so that I don't get tired which will affect future development :smile: ).

**Comment by SAbboushi** [comment]
>> can I send it to your account on the AutoHotkey Forums?
Yes please

**Comment by zero-plusplus** [comment]
I sent a test message first because it was my first time using private messaging and I wasn't sure if I was using it correctly. I will send you my email address tomorrow when I confirm that my test message was sent correctly.

Sorry for one more step.

**Comment by SAbboushi** [comment]
Yes I received it.  It would not allow me to respond: the error is "Some users couldn’t be added as they have disabled private message receipt."

**Comment by zero-plusplus** [comment]
Oops.

Please send me your email address from you if you would like.
Looking at my inbox, it seems that my account was able to receive private messages from you.

**Comment by SAbboushi** [comment]
Even though "zero-plusplus" is on my list of "Friends", whenever I try to add "zero-plusplus", I receive message "Some users couldn’t be added as they have disabled private message receipt."  As test, I was able to add "Zero-Blade" which then is listed under "Recipients".
![image](https://github.com/user-attachments/assets/45ad9bc2-bdf4-4bd4-9c6a-bbbec97469e2)

You said: "Looking at my inbox, it seems that my account was able to receive private messages from you."
I believe I remember something about the forum software being updated or changed, so that may explain why I was able to send you a message in 2020 but cannot send one now.  I suggest you temporarily enable PM until you after you receive my message:
![image](https://github.com/user-attachments/assets/2eb12fd7-2d1b-4998-a005-cc2ebd31b0c2)

**Comment by zero-plusplus** [comment]
I checked and confirmed that the setting is indeed “No”.
I changed it to “Yes” and clicked “Submit”, so the setting should be OK.

So the test message I just posted should be sent to you.

Thanks for letting me know.

**Comment by zero-plusplus** [comment]
@SAbboushi 

Sorry, I didn't check the AutoHotkey Forum and noticed this late. I sent a test message to your email address, please check it.

I'll notify you here as well, just in case.
